synapse/handlers/_base.py

```diff
@@ -57,10 +57,24 @@ def ratelimit(self, requester):
         time_now = self.clock.time()
         user_id = requester.user.to_string()
 
+        # Disable rate limiting of users belonging to any AS that is configured
+        # not to be rate limited in its registration file (rate_limited: true|false).
+        # The AS user itself is never rate limited.
+
         app_service = self.store.get_app_service_by_user_id(user_id)
         if app_service is not None:
             return  # do not ratelimit app service senders
 
+        should_rate_limit = True
+
+        for service in self.store.get_app_services():
+            if service.is_interested_in_user(user_id):
```

Comment:
This check looks like it could be relatively expensive and is on a vaguely hot path. (Its also wrong if multiple AS's register non-exclusive interest in it).

I'd rather we did this by checking if the authenticated thing actually doing the sending was an AS. We can probably do this by adding a field to the requester type when we authenticate the sender in [auth module](https://github.com/matrix-org/synapse/blob/v0.18.1/synapse/api/auth.py#L606-L609) and checking that field here.